`zigmaster` branch doesn't works on `0.12.0-dev.2046+d3a163f86`

branch: `zigmaster`
zig version: `0.12.0-dev.2046+d3a163f86 `

output:
```
/zig/p/122071524b599acc6268eb02f3aaefbb72761276b7ebf05428ab57e6a0551caf312c/build.zig:19:10: error: no field named 'source_file' in struct 'Build.Module.CreateOptions'
        .source_file = .{ .path = "src/main.zig" },
         ^~~~~~~~~~~
/zig/master/lib/std/Build/Module.zig:140:27: note: struct declared here
pub const CreateOptions = struct {
                          ^~~~~~
referenced by:
    runBuild__anon_15995: /zig/master/lib/std/Build.zig:1851:37
    dependencyInner__anon_14600: /zig/master/lib/std/Build.zig:1831:29
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
/zig/p/12208dc0796bffa6ea9edb60193108e34f835f35fc152e853717aca2b13ba17f3be2/build.zig:9:32: error: no field named 
'source_file' in struct 'Build.Module.CreateOptions'
    _ = b.addModule("xev", .{ .source_file = .{ .path = "src/main.zig" } });
                               ^~~~~~~~~~~
/zig/master/lib/std/Build/Module.zig:140:27: note: struct declared here
pub const CreateOptions = struct {
                          ^~~~~~
```

PR that fixes it: https://github.com/rsepassi/zigcoro/pull/17

--- a/README.md
+++ b/README.md
@@ -10,7 +10,7 @@ Supports async IO via [`libxev`][libxev].
 
 Branch `main` tested against Zig v0.11.0.
 
-Branch `zigmaster` tested against Zig v0.12.0-dev.1372+8457439a8.
+Branch `zigmaster` tested against Zig v0.12.0-dev.2046+d3a163f86.
 
 Coroutines supported on Windows `x86_64`, Linux {`x86_64`, `aarch64`, `riscv64`}, and Mac {`x86_64`, `aarch64`}.
 

--- a/benchmark.zig
+++ b/benchmark.zig
@@ -28,7 +28,7 @@ fn contextSwitchBm() !void {
         // warmup
         num_bounces = 100_000;
         {
-            var test_coro = try libcoro.xasync(testFn, .{}, stack);
+            const test_coro = try libcoro.xasync(testFn, .{}, stack);
             for (0..num_bounces) |_| {
                 libcoro.xresume(test_coro);
             }
@@ -37,7 +37,7 @@ fn contextSwitchBm() !void {
 
         num_bounces = 20_000_000;
         {
-            var test_coro = try libcoro.xasync(testFn, .{}, stack);
+            const test_coro = try libcoro.xasync(testFn, .{}, stack);
 
             const start = std.time.nanoTimestamp();
             for (0..num_bounces) |_| {
@@ -118,12 +118,12 @@ fn ncorosBm(num_coros: usize) !void {
     var coros = try allocator.alloc(libcoro.Frame, num_coros);
     defer allocator.free(coros);
 
-    var buf = try allocator.alloc(u8, num_coros * 1024 * 4);
+    const buf = try allocator.alloc(u8, num_coros * 1024 * 4);
     var fba = std.heap.FixedBufferAllocator.init(buf);
     const alloc2 = fba.allocator();
 
     for (0..num_coros) |i| {
-        var stack = try libcoro.stackAlloc(alloc2, null);
+        const stack = try libcoro.stackAlloc(alloc2, null);
         const coro = try libcoro.xasync(suspendRepeat, .{}, stack);
         coros[i] = coro.frame();
     }

--- a/build.zig
+++ b/build.zig
@@ -30,7 +30,7 @@ pub fn build(b: *std.Build) !void {
             .target = target,
             .optimize = optimize,
         });
-        coro_test.addModule("libcoro", coro);
+        coro_test.root_module.addImport("libcoro", coro);
         coro_test.linkLibC();
 
         const internal_test = b.addTest(.{
@@ -39,7 +39,7 @@ pub fn build(b: *std.Build) !void {
             .target = target,
             .optimize = optimize,
         });
-        internal_test.addModule("libcoro_options", coro_options_module);
+        internal_test.root_module.addImport("libcoro_options", coro_options_module);
         internal_test.linkLibC();
 
         // Test step
@@ -55,8 +55,8 @@ pub fn build(b: *std.Build) !void {
             .target = target,
             .optimize = optimize,
         });
-        aio_test.addModule("libcoro", coro);
-        aio_test.addModule("xev", xev);
+        aio_test.root_module.addImport("libcoro", coro);
+        aio_test.root_module.addImport("xev", xev);
         aio_test.linkLibC();
 
         // Test step
@@ -71,7 +71,7 @@ pub fn build(b: *std.Build) !void {
             .root_source_file = .{ .path = "benchmark.zig" },
             .optimize = .ReleaseFast,
         });
-        bench.addModule("libcoro", coro);
+        bench.root_module.addImport("libcoro", coro);
         bench.linkLibC();
         const bench_run = b.addRunArtifact(bench);
         if (b.args) |args| {

--- a/build.zig
+++ b/build.zig
@@ -16,8 +16,8 @@ pub fn build(b: *std.Build) !void {
     coro_options.addOption(usize, "debug_log_level", debug_log_level);
     const coro_options_module = coro_options.createModule();
     const coro = b.addModule("libcoro", .{
-        .source_file = .{ .path = "src/main.zig" },
-        .dependencies = &[_]std.Build.ModuleDependency{
+        .root_source_file = .{ .path = "src/main.zig" },
+        .imports = &.{ 
             .{ .name = "xev", .module = xev },
             .{ .name = "libcoro_options", .module = coro_options_module },
         },

--- a/build.zig
+++ b/build.zig
@@ -69,6 +69,7 @@ pub fn build(b: *std.Build) !void {
         const bench = b.addExecutable(.{
             .name = "benchmark",
             .root_source_file = .{ .path = "benchmark.zig" },
+            .target = target,
             .optimize = .ReleaseFast,
         });
         bench.root_module.addImport("libcoro", coro);

--- a/build.zig.zon
+++ b/build.zig.zon
@@ -3,8 +3,8 @@
     .version = "0.0.0",
     .dependencies = .{
         .libxev = .{
-            .url = "https://api.github.com/repos/mitchellh/libxev/tarball/1b46c2d",
-            .hash = "12208dc0796bffa6ea9edb60193108e34f835f35fc152e853717aca2b13ba17f3be2",
+            .url = "git+https://github.com/mitchellh/libxev#main",
+            .hash = "12203de3dbf9c235903c71122448ef031622ba8d64b250205934e2a65018b92b9717",
         },
     },
     .paths = .{

--- a/build.zig.zon
+++ b/build.zig.zon
@@ -3,8 +3,8 @@
     .version = "0.0.0",
     .dependencies = .{
         .libxev = .{
-            .url = "git+https://github.com/mitchellh/libxev#main",
-            .hash = "12203de3dbf9c235903c71122448ef031622ba8d64b250205934e2a65018b92b9717",
+            .url = "git+https://github.com/mitchellh/libxev#74bc7aea4a8f88210f0ad4215108613ab7e7af1a",
+            .hash = "122029743e5d96aa1b57a1b99ff58bf13ff9ed6d8f624ac3ae8074062feb91c5bd8d",
         },
     },
     .paths = .{

--- a/src/coro.zig
+++ b/src/coro.zig
@@ -92,7 +92,7 @@ pub fn xasync(func: anytype, args: anytype, stack: anytype) !FrameT(func, .{ .Ar
         .ArgsT = @TypeOf(args),
     });
     const framet = try FrameType.init(args, stack_info.stack, stack_info.owned);
-    var frame = framet.frame();
+    const frame = framet.frame();
     xresume(frame);
     return FrameType.wrap(frame);
 }
@@ -193,7 +193,7 @@ const Coro = struct {
 
     fn initFromStack(func: *const fn () void, stack: *Stack, owns_stack: bool, storage: ?*anyopaque) !Frame {
         try StackOverflow.setMagicNumber(stack.full);
-        var coro = try stack.push(Coro);
+        const coro = try stack.push(Coro);
         const base_coro = try base.Coro.init(&runcoro, stack.remaining());
         coro.* = @This(){
             .func = func,
@@ -322,7 +322,7 @@ const CoroT = struct {
                 owns_stack: bool,
             ) !Self {
                 var s = Stack.init(stack);
-                var inner = try s.push(InnerStorage);
+                const inner = try s.push(InnerStorage);
                 inner.* = .{
                     .args = args,
                 };
@@ -648,7 +648,7 @@ test "basic suspend and resume" {
     defer allocator.free(stack);
 
     testSetIdx(0);
-    var test_coro = try Coro.init(testFn, stack, false, null);
+    const test_coro = try Coro.init(testFn, stack, false, null);
 
     testSetIdx(1);
     try std.testing.expectEqual(test_coro.status, .Start);
@@ -690,7 +690,7 @@ test "with values" {
     const allocator = std.testing.allocator;
     const stack = try stackAlloc(allocator, null);
     defer allocator.free(stack);
-    var coro = try Coro.init(Test.coroWrap, stack, false, @ptrCast(&storage));
+    const coro = try Coro.init(Test.coroWrap, stack, false, @ptrCast(&storage));
 
     try std.testing.expectEqual(storage.x.*, 0);
     xresume(coro);
@@ -744,7 +744,7 @@ test "iterator" {
     const stack = try stackAlloc(allocator, null);
     defer allocator.free(stack);
 
-    var x: usize = 1;
+    const x: usize = 1;
     var coro = try Iter.init(.{x}, stack, false);
     var yielded: usize = undefined;
     yielded = Iter.xnextStart(coro); // first resume takes no value

--- a/src/test_aio.zig
+++ b/src/test_aio.zig
@@ -19,6 +19,9 @@ const AioTest = struct {
         var tp = try allocator.create(xev.ThreadPool);
         var loop = try allocator.create(xev.Loop);
         var exec = try allocator.create(aio.Executor);
+        _ = &tp;
+        _ = &loop;
+        _ = &exec;
         tp.* = xev.ThreadPool.init(.{});
         loop.* = try xev.Loop.init(.{ .thread_pool = tp });
         exec.* = aio.Executor.init(loop);
@@ -98,14 +101,14 @@ fn sleepTask() !void {
         null,
     );
     defer env.allocator.free(stack);
-    var sleep1 = try aio.xasync(sleep, .{10}, stack);
+    const sleep1 = try aio.xasync(sleep, .{10}, stack);
 
     const stack2 = try libcoro.stackAlloc(
         env.allocator,
         null,
     );
     defer env.allocator.free(stack2);
-    var sleep2 = try aio.xasync(sleep, .{20}, stack2);
+    const sleep2 = try aio.xasync(sleep, .{20}, stack2);
 
     const after = try aio.xawait(sleep1);
     const after2 = try aio.xawait(sleep2);
@@ -231,11 +234,11 @@ fn udpMain() !void {
 
     const stack1 = try libcoro.stackAlloc(env.allocator, stack_size);
     defer env.allocator.free(stack1);
-    var server_co = try aio.xasync(udpServer, .{&info}, stack1);
+    const server_co = try aio.xasync(udpServer, .{&info}, stack1);
 
     const stack2 = try libcoro.stackAlloc(env.allocator, stack_size);
     defer env.allocator.free(stack2);
-    var client_co = try aio.xasync(udpClient, .{&info}, stack2);
+    const client_co = try aio.xasync(udpClient, .{&info}, stack2);
 
     try aio.xawait(server_co);
     try aio.xawait(client_co);
@@ -272,11 +275,11 @@ fn asyncMain() !void {
 
     const stack = try libcoro.stackAlloc(env.allocator, stack_size);
     defer env.allocator.free(stack);
-    var co = try aio.xasync(asyncTest, .{&nstate}, stack);
+    const co = try aio.xasync(asyncTest, .{&nstate}, stack);
 
     const stack2 = try libcoro.stackAlloc(env.allocator, stack_size);
     defer env.allocator.free(stack2);
-    var nco = try aio.xasync(asyncNotifier, .{&nstate}, stack2);
+    const nco = try aio.xasync(asyncNotifier, .{&nstate}, stack2);
 
     try aio.xawait(co);
     try aio.xawait(nco);

